mmdet/core/general_data/general_data.py

```diff
@@ -257,61 +288,37 @@ def detach(self):
     def numpy(self):
         """Apply same name function to all tensors in results field."""
         new_results = self.new_results()
-        for k, v in self.results_field.items():
+        for k, v in self.items():
             if isinstance(v, torch.Tensor):
                 v = v.detach().cpu().numpy()
             new_results[k] = v
         return new_results
 
-    def format_results(self):
-        """The protocols for serializing the value in `results_field`,
-        Generally the value would be converted into a format suitable for
-        multi-GPU synchronization.
-
-        Returns:
-            dict
-        """
-        results_dict = {}
-
-        for name, results in self.results_field.items():
-            if hasattr(results, 'format_results'):
-                update_dict = results.format_results()
-            else:
-                update_dict = {name: results}
-
-            assert not set(update_dict.keys()) & \
-                set(results_dict.keys()), f'Find' \
-                f' duplicate keys ' \
-                f'{set(update_dict.keys()) & set(results_dict.keys())} ' \
-                f'when format the {self}'
-            results_dict.update(update_dict)
-
-        return results_dict
-
     def __nice__(self):
         repr = '\n \n  META INFORMATION \n'
-        for k, v in self.meta_info_field.items():
+        for k, v in self.meta_items():
             repr += f'{k}: {v} \n'
-        repr += '\n   PREDICTIONS \n'
-        for k, v in self.results_field.items():
+        repr += '\n   DATA FIELDS \n'
+        for k, v in self.items():
             if isinstance(v, (torch.Tensor, np.ndarray)):
                 repr += f'shape of {k}: {v.shape} \n'
             else:
                 repr += f'{k}: {v} \n'
         return repr + '\n'
 
 
-class InstanceResults(Results):
-    """Subclass of results. All value in `results_field` should has same
-    length.
+class Instances(GeneralData):
```

Comment:
Since we will have PerPixelData and SampleLevelData, name this class with InstanceData would be better.